Codd doesn't support mtl >= 2.3

```
Building library for codd-0.1.3..
[ 1 of 29] Compiling Codd.Logging     ( src/Codd/Logging.hs, dist/build/Codd/Logging.o, dist/build/Codd/Logging.dyn_o )

src/Codd/Logging.hs:20:51: error:
    Module ‘Control.Monad.Reader’ does not export ‘when’
   |
20 |                                                 , when
   |                                                   ^^^^
```

In [mtl-2.2.2](https://hackage.haskell.org/package/mtl-2.2.2/docs/Control-Monad-Reader.html), `Control.Monad.Reader` re-exported `Control.Monad`, so `when` could be imported directly from it. In [mtl-2.3](https://hackage.haskell.org/package/mtl-2.3/docs/Control-Monad-Reader.html) that's been removed.

I haven't tried to fix this and see if anything else breaks, so there might be more to do.

--- a/src/Codd/Logging.hs
+++ b/src/Codd/Logging.hs
@@ -14,10 +14,10 @@ module Codd.Logging
     , runCoddLoggerLevelFilter
     ) where
 
+import           Control.Monad                  ( when )
 import           Control.Monad.Reader           ( MonadReader(..)
                                                 , MonadTrans(..)
                                                 , ReaderT(..)
-                                                , when
                                                 )
 import           Control.Monad.Trans.Resource   ( MonadThrow )
 import           Data.Text                      ( Text )

